### ISSUE TITLE
reproducibility on SSL case

### ISSUE THREAD

**Issue by luoyuchenmlcv** [ruoxi-jia-group/ASSET#1]
Thanks for sharing your amazing work!

Could you release code for SSL? 

What is your details about backdoor attack SSL on C-brd, C-squ on Cifar10? how does 0.5% comes? There are 5000 samples in each class for Cifar10, how do you distribute the 250 poison sample among classes? 
![image](https://github.com/ruoxi-jia-group/ASSET/assets/102636806/719b0c90-98c0-4168-a821-8bc8d14490a3)

**Comment by Lin23508** [comment]
Thanks for sharing your amazing work!
Could you release the code in the checkpoint (checkpoint/cifar10_backdoor_0.1_resnet18_tar2.pth)? 
The attack model I trained myself couldn't achieve the desired effect.
Thanks for your support!

**Comment by pmzzs** [comment]
> Thanks for sharing your amazing work! Could you release the code in the checkpoint (checkpoint/cifar10_backdoor_0.1_resnet18_tar2.pth)? The attack model I trained myself couldn't achieve the desired effect. Thanks for your support!

Here is the .pth file for that https://drive.google.com/file/d/1PC9RrjEbIfsMwP75PLy2YdzSo5205wV5/view?usp=sharing

**Comment by pmzzs** [comment]
> Thanks for sharing your amazing work!
> 
> Could you release code for SSL?
> 
> What is your details about backdoor attack SSL on C-brd, C-squ on Cifar10? how does 0.5% comes? There are 5000 samples in each class for Cifar10, how do you distribute the 250 poison sample among classes? ![image](https://private-user-images.githubusercontent.com/102636806/268430648-719b0c90-98c0-4168-a821-8bc8d14490a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1Mjg3NjEsIm5iZiI6MTcwNTUyODQ2MSwicGF0aCI6Ii8xMDI2MzY4MDYvMjY4NDMwNjQ4LTcxOWIwYzkwLTk4YzAtNDE2OC1hODIxLThiYzhkMTQ0OTBhMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExN1QyMTU0MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZGNkNzc0NjU3YzczYWEwZTUyYTZkMWZjMjdmOGI1YTYwNTNhYjQ2NDJlMGVjZTIyOTkwMzUyZTA4NWE1ZTQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eAP8_ahlADDNgHFxrIwYEhCTTpD_s_4kzmpgYZws0IM)

For SSL, you can simply change the model with an SSL pre-trained one without any other change. For the 0.5% poison ratio, we only select 250 images from the target class.

**Comment by luoyuchenmlcv** [comment]
> > Thanks for sharing your amazing work!
> > Could you release code for SSL?
> > What is your details about backdoor attack SSL on C-brd, C-squ on Cifar10? how does 0.5% comes? There are 5000 samples in each class for Cifar10, how do you distribute the 250 poison sample among classes? ![image](https://private-user-images.githubusercontent.com/102636806/268430648-719b0c90-98c0-4168-a821-8bc8d14490a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1Mjg3NjEsIm5iZiI6MTcwNTUyODQ2MSwicGF0aCI6Ii8xMDI2MzY4MDYvMjY4NDMwNjQ4LTcxOWIwYzkwLTk4YzAtNDE2OC1hODIxLThiYzhkMTQ0OTBhMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExN1QyMTU0MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZGNkNzc0NjU3YzczYWEwZTUyYTZkMWZjMjdmOGI1YTYwNTNhYjQ2NDJlMGVjZTIyOTkwMzUyZTA4NWE1ZTQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eAP8_ahlADDNgHFxrIwYEhCTTpD_s_4kzmpgYZws0IM)
> 
> For SSL, you can simply change the model with an SSL pre-trained one without any other change. For the 0.5% poison ratio, we only select 250 images from the target class.

For Cifar10 with SSL-backdoor attack, there are only 10 class, as mentioned in your paper, you poison half in-class samples, shoud not it be 2500 poison samples in this case? Also, for SSL backdoor denfense, did you use label information for CE loss on your reserved clean set other than minimizing the variance loss? Thank you!

**Comment by Lin23508** [comment]
Thanks a lot!

從 iOS 版 Outlook<https://aka.ms/o0ukef> 寄出
________________________________
寄件者: Mz Pan ***@***.***>
寄件日期: Wednesday, January 17, 2024 4:53:09 PM
收件者: ruoxi-jia-group/ASSET ***@***.***>
副本: Lin23508 ***@***.***>; Comment ***@***.***>
主旨: Re: [ruoxi-jia-group/ASSET] reproducibility on SSL case (Issue #1)


Thanks for sharing your amazing work! Could you release the code in the checkpoint (checkpoint/cifar10_backdoor_0.1_resnet18_tar2.pth)? The attack model I trained myself couldn't achieve the desired effect. Thanks for your support!

Here is the .pth file for that https://drive.google.com/file/d/1PC9RrjEbIfsMwP75PLy2YdzSo5205wV5/view?usp=sharing

**Comment by pmzzs** [comment]
> > > Thanks for sharing your amazing work!
> > > Could you release code for SSL?
> > > What is your details about backdoor attack SSL on C-brd, C-squ on Cifar10? how does 0.5% comes? There are 5000 samples in each class for Cifar10, how do you distribute the 250 poison sample among classes? ![image](https://private-user-images.githubusercontent.com/102636806/268430648-719b0c90-98c0-4168-a821-8bc8d14490a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1Mjg3NjEsIm5iZiI6MTcwNTUyODQ2MSwicGF0aCI6Ii8xMDI2MzY4MDYvMjY4NDMwNjQ4LTcxOWIwYzkwLTk4YzAtNDE2OC1hODIxLThiYzhkMTQ0OTBhMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExN1QyMTU0MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZGNkNzc0NjU3YzczYWEwZTUyYTZkMWZjMjdmOGI1YTYwNTNhYjQ2NDJlMGVjZTIyOTkwMzUyZTA4NWE1ZTQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eAP8_ahlADDNgHFxrIwYEhCTTpD_s_4kzmpgYZws0IM)
> > 
> > 
> > For SSL, you can simply change the model with an SSL pre-trained one without any other change. For the 0.5% poison ratio, we only select 250 images from the target class.
> 
> For Cifar10 with SSL-backdoor attack, there are only 10 class, as mentioned in your paper, you poison half in-class samples, shoud not it be 2500 poison samples in this case? Also, for SSL backdoor denfense, did you use label information for CE loss on your reserved clean set other than minimizing the variance loss? Thank you!

Thank you for pointing out the mistake, it should be 2500 poison samples indeed. For all learning paradigms, also including SSL, the reserved clean set is always been minimized through variance loss. The CE loss will only be used in the training dataset when label information is available.
